packages/url_launcher/url_launcher_macos/example/macos/RunnerTests/RunnerTests.swift

```diff
@@ -6,19 +6,121 @@ import FlutterMacOS
 import XCTest
 import url_launcher_macos
 
+/// A stub to simulate the system Url handler.
+class StubWorkspace: SystemURLHandler {
+
+  var isSuccessful = true
+
+  func open(_ url: URL) -> Bool {
+    return isSuccessful
+  }
+
+  func urlForApplication(toOpen: URL) -> URL? {
+    return toOpen
+  }
+}
+
 class RunnerTests: XCTestCase {
-  func testCanLaunch() throws {
+
+  func testCanLaunchSuccessReturnsTrue() throws {
     let plugin = UrlLauncherPlugin()
+
     let call = FlutterMethodCall(
       methodName: "canLaunch",
       arguments: ["url": "https://flutter.dev"])
-    var canLaunch: Bool?
+
+    plugin.handle(
+      call,
+      result: { (result: Any?) -> Void in
+        XCTAssertEqual(result as? Bool, true)
+      })
```

Comment:
Removing `result` calls from the implementation would not fail any of these tests as you have written them, because the only assertion is in the callback.

The usual pattern for our tests is that you create [an expectation](https://developer.apple.com/documentation/xctest/asynchronous_tests_and_expectations/testing_asynchronous_operations_with_expectations?language=objc) that is `fulfill`ed in the `result` callback.

Reply:
Thank you for the advice. All the tests now should fulfill an expectation in the completion handler.